sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -1268,3 +1461,140 @@ def display_data(self):
                                    tableSpec)
       res['table'] = DisplayDataItem(tableSpec, label='Table')
     return res
+
+
+@experimental()
+class PassThroughThenCleanup(PTransform):
+  """A PTransform that invokes a DoFn after the input PCollection has been
+    processed.
+  """
+  def __init__(self, cleanup_dofn):
+    self.cleanup_dofn = cleanup_dofn
+
+  def expand(self, input):
+    class PassThrough(beam.DoFn):
+      def process(self, element):
+        yield element
+
+    output = input | beam.ParDo(PassThrough()).with_outputs('cleanup_signal',
+                                                            main='main')
+    main_output = output['main']
+    cleanup_signal = output['cleanup_signal']
+
+    _ = (input.pipeline
+         | beam.Create([None])
+         | beam.ParDo(self.cleanup_dofn, beam.pvalue.AsSingleton(
+             cleanup_signal)))
+
+    return main_output
+
+
+@experimental()
+class ReadFromBigQuery(PTransform):
```

Comment:
This is looking great. I've discussed with Cham, and let's rename this as `_ReadFromBigQuery` (with underscore) to prevent users from picking it up before we have tested it. We have some tests that we'll run on it, and once we're confident of performance/functionality, we can remove the underscore, and rename to `ReadFromBigQuery`. WDYT?

Reply:
I'm totally fine with it. 
Once this PR is merged, I'm going to make changes to the Chicago Taxi Example, so that the example would use this transform. That would be a great opportunity to check stability and measure performance of the transform

Reply:
Yes, that will be great to compare the Native Dataflow source with the new Beam source.